src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -261,14 +276,48 @@ public String getStorageAccountName() {
         return storageAccountName;
     }
 
+    public static String getStorageAccountName(final String type, final String newName, final String existName) {
+        //type maybe null in this version, so we can guess according to whether newName is blank or not
+        if (StringUtils.isBlank(type) && StringUtils.isNotBlank(newName)
+                || StringUtils.isNotBlank(type) && type.equalsIgnoreCase("new")) {
+            return newName;
+        }
+        return existName;
+    }
+
+    public String getStorageAccountNameReferenceType() {
+        return storageAccountNameReferenceType;
+    }
+
     public void setStorageAccountName(final String storageAccountName) {
         this.storageAccountName = storageAccountName;
     }
 
+    public String getNewStorageAccountName() {
+        //backward compatibility, the old version plugin only have storageAccountName
+        if (StringUtils.isBlank(newStorageAccountName) && StringUtils.isBlank(existStorageAccountName)
```

Comment:
According to [Jenkins wiki](https://wiki.jenkins-ci.org/display/JENKINS/Hint+on+retaining+backward+compatibility), it's recommended to put backward compatibility code in `readResolve()` method?